.github/workflows/nvm.yml

```
@@ -5,9 +5,47 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - run: node --version
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v4
       - run: |
           nvm install --latest-npm --no-progress
-          echo "::set-env name=PATH::${NVM_BIN}:${PATH}"
+          echo "$(dirname $(which node))" >> $GITHUB_PATH
         shell: bash --login {0}
       - run: node --version
+
+      # Optional: Cache npm dependencies
+      - name: Get npm cache directory
+        id: npm-cache-dir
+        shell: bash
+        run: echo "dir=$(npm config get cache)" >> $GITHUB_OUTPUT
+      - uses: actions/cache@v4
+        id: npm-cache
+        with:
+          path: ${{ steps.npm-cache-dir.outputs.dir }}
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-
+
+      # Optional: Cache Yarn v1 dependencies
+      - name: Get yarn cache directory path
+        id: yarn-cache-dir-path
```

[suggest] Let's keep consistency with `Get npm cache directory` and `npm-cache-dir`.

```suggestion
      - name: Get yarn cache directory
        id: yarn-cache-dir
```